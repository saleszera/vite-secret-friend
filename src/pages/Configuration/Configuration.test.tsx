import { render } from '@testing-library/react';
import { RecoilRoot } from 'recoil';
import { vi, describe, it, expect } from 'vitest';

import { Configuration } from '.';

const mockNavigation = vi.fn();

vi.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigation,
}));

describe('Configuration page', () => {
  it('should be able correctly render', () => {
    const { container } = render(
      <RecoilRoot>
        <Configuration />
      </RecoilRoot>
    );

    expect(container).toMatchSnapshot();
  });
});
