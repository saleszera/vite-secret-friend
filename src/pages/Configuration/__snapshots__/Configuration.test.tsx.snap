// Vitest Snapshot v1

exports[`Configuration page > should be able correctly render 1`] = `
<div>
  <div
    class="sc-bczRLJ jvcWzg"
  >
    <section>
      <h2>
        configuration:start
      </h2>
      <form>
        <div
          class="sc-dkzDqf FPKVq"
        >
          <input
            placeholder="configuration:insertParticipantsNames"
            type="text"
            value=""
          />
          <button
            disabled=""
            type="submit"
          >
            configuration:add
          </button>
        </div>
      </form>
      <ul />
      <footer
        class="sc-gsnTZi eVxbKh"
      >
        <button
          disabled=""
          type="button"
        >
          configuration:startGame
        </button>
        <img
          alt="shopping bags"
          src="/src/assets/images/bags.svg"
        />
      </footer>
    </section>
  </div>
</div>
`;
